asyn/asynPortDriver/asynPortDriver.cpp

```diff
@@ -146,8 +147,11 @@ asynStatus paramList::createParam(const char *name, asynParamType type, int *ind
   * \return Returns asynParamNotFound if name is not found in the parameter list. */
 asynStatus paramList::findParam(const char *name, int *index)
 {
-    for (*index=0; *index<this->nVals; (*index)++) {
-        if (this->vals[*index]->nameEquals(name)) return asynSuccess;
+    for (size_t i=0; i<this->vals.size(); i++) {
```

Comment:
vector is the one container where this makes no practical difference.  It is a smaller change, and easier to validate, to keep the indexed access.